src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -306,8 +304,12 @@ internal static ExecutionStrategy GetExecutionStrategy(WorkItem work)
                 work.Test is TestFixture && work.Context.ParallelScope.HasFlag(ParallelScope.Fixtures))
                     return ExecutionStrategy.Parallel;
 
-            // If all else fails, run on same thread
-            return ExecutionStrategy.Direct;
+            // There is no scope specified either on the item itself or in the context.
+            // In that case, simple work items are test cases and just run on the same
+            // thread, while composite work items and teardowns are non-parallel.
+            return work is SimpleWorkItem
+                ? ExecutionStrategy.Direct
+                : ExecutionStrategy.NonParallel;
```

Comment:
The dispatcher is responsible for deciding how to run a work item. Different dispatchers do it differently and in principle the work item doesn't know what those strategy choices are.

it seems more reasonable to me for the. dispatcher to know about work items since it operates on them, but I agree that a test on type is ugly.

in thus case the check is on whether the item might have children. I could use work.Test.IsSuite or give the item itself an IsComposite member.